ibmdbpy/frame.py

```diff
@@ -2316,3 +2323,106 @@ def check_numeric_columns(idaobject):
+        """
+        
+        Parameters
+        -------
+        columns: list of eligible column names
+        logic: str, optional, "any" by default. If logic is set to "any" then all rows which contain a NaN value in any 
+        (id est at least one) of the cited columns is deleted, this is a union condition; 
+        if logic is set to "all", then only rows containing null values in all the fields are deleted, this is an intersection condition.         
+        inplace: bool, optional, False by default. If True, then the underlying table is physically modified,
+        if False, then the original objects remain unaffected, a copy of the IdaDataFrame is made and modified. 
+        
+        Returns
+        -------
+        If inplace is True, the original table is modified and the IdaDataFrame will be modified accordingly. No return.
+        If inplace is False, a new IdaDataFrame is returned, the original table is not affected. This new IdaDataFrame points to a 
+        new table which has been created according to the user defined criterions on columns.
+        
+        Examples
+        -------
+        idadf.delete_na(["COL_1", "COL_2"], inplace = True)
+        # each row of the original table is physically deleted if there is a NULL value in one of the listed columns
+        new_idadf = idadf.delete_na(["COL_1", "COL_2"], union = False)
+        # a new IdaDataFrame is created, rows will be selected only if all the listed columns have a NULL value.
+        """
+        
+        # Check if list columns is not empty
+        if len(columns)<1:
+            raise IdaDataFrameError("You must specify the columns as a list of eligible names")
+        # Check if column names are eligible
+        for column_name in columns:
+            if not isinstance(column_name, six.string_types):
+                raise TypeError("%s is not of string type"%(column_name))
+            if column_name not in self.columns:
+                raise ValueError("%s refers to a columns that doesn't exists in self"%(column_name))
+        
+        idadb = self._idadb
+        
+        if inplace == True:
+            # Ask for confirmation
+            print("The table %s will be physically modified." %self.tablename)
+            answer = input('Do you want to proceed? [y/n]')
+            if answer == 'n':
+                print('Then use the default option inplace = False')
+                return None
+            if answer == 'y':
+                print("Any IdaDataFrame pointing this table might be modified accordingly.")
+            else:
+                print("User input should be either 'y' or 'n'.")
+                return None
+            # Write DELETE query on original table
+            tablename = self.tablename
+            query = 'DELETE FROM %s WHERE "%s" IS NULL'%(tablename, columns[0])
+            
+            if len(columns)>1:
+                if logic == "any":
+                    for i in range(1,len(columns)):
```

Comment:
You can just do

```suggestion
                    for column in columns:
```

to iterate over the column names. (Happens multiple times in this function)

Reply:
I want to exclude the first item, hence the range(1,...)

Reply:
Ah, then use `for column in columns[1:]`.

Reply:
changed